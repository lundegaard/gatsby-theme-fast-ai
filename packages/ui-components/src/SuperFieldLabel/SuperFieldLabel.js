import React from 'react';
import PropTypes from 'prop-types';

import Label from '../Label';
import useBlockingEffect from '../hooks/useBlockingEffect';
import { useSuperFieldContext } from '../SuperField';

const SuperFieldLabel = ({ children, alwaysShrank }) => {
	const {
		id,
		onLabelShrank,
		onLabelExpand,
		isFocused,
		isInputFilledIn,
		isLabelShrank,
	} = useSuperFieldContext();

	useBlockingEffect(() => {
		if (isInputFilledIn || isFocused || alwaysShrank) {
			onLabelShrank();
		} else {
			onLabelExpand();
		}
	}, [isFocused, isInputFilledIn]);

	return (
		<Label
			htmlFor={id}
			sx={{
				position: 'absolute',
				top: 0,
				left: 0,
				transition: 'color 200ms ease, transform 200ms ease 0ms',
				transform: isLabelShrank
					? 'translate(0, 0) scale(0.75)'
					: 'translate(0, 16px)',
				transformOrigin: 'top left',
				color: 'inherit',
			}}
		>
			{children}
		</Label>
	);
};
SuperFieldLabel.propTypes = {
	alwaysShrank: PropTypes.bool,
	children: PropTypes.node,
};

export default SuperFieldLabel;
