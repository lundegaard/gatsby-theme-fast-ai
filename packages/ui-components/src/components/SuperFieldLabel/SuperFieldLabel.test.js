import React from 'react';
import { shallow } from 'enzyme';

import SuperFieldLabel from './SuperFieldLabel';

describe('<SuperFieldLabel />', () => {
	const requiredProps = {};

	it('matches snapshot when shallowly rendered', () => {
		const wrapper = shallow(<SuperFieldLabel {...requiredProps} />);
		expect(wrapper).toMatchSnapshot();
	});
});
