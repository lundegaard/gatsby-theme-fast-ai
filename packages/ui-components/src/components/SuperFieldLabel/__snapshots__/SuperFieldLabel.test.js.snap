// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SuperFieldLabel /> matches snapshot when shallowly rendered 1`] = `
<ForwardRef
  htmlFor={null}
  sx={
    Object {
      "color": "inherit",
      "left": 0,
      "position": "absolute",
      "top": 0,
      "transform": "translate(0, 16px)",
      "transformOrigin": "top left",
      "transition": "color 200ms ease, transform 200ms ease 0ms",
    }
  }
/>
`;
